### PR TITLE
openstack: cache auth tokens

### PR DESCRIPTION
**Pull request by dachary**
http://tracker.ceph.com/issues/16893

### CODE REVIEW

File: teuthology/nuke.py [hunk 1]
```diff
@@ -442,16 +442,16 @@ def stale_openstack_instances(ctx, instances, locked_nodes):
 
 
 def openstack_delete_volume(id):
-    sh("openstack volume delete " + id + " || true")
+    OpenStack().run("volume delete " + id + " || true")
```
Comment (zmc) [inline teuthology/nuke.py]: Why is this call using an instance where the other one in this module is not?

Comment (dachary) [inline teuthology/nuke.py]: My bad: in an earlier version I had run as a static function but it's no longer the case.

**Comment by smithfarm** [comment]
@dachary Rebase?

**Comment by dachary** [comment]
@smithfarm rebased

**Comment by dmick** [comment]
Two of these are "missing -q to the call params", but the last one is "used to be sh(), now is OpenStack.run()" and I'm not immediately sure how to mock that.

**Comment by dachary** [comment]
@dmick which one are missing -q ? None of them should use it now that the run method exists, because it adds --quiet to all.

**Comment by dmick** [comment]
see the test results

**Comment by dachary** [comment]
@dmick oh, right :-) I thought you were refering to parts of the code where -q was missing. Fixed & repushed !

**Comment by dachary** [comment]
<pre>
        def sh(cmd):
            if 'volume show' in cmd:
                return volume_show
    
        with patch.multiple(
                nuke,
                sh=sh,
                openstack_delete_volume=DEFAULT,
                ) as m:
            nuke.stale_openstack_volumes(ctx, volume_list)
>           m['openstack_delete_volume'].assert_called_with(id)
</pre>
the error is because it does not mock the function it should

**Comment by dmick** [comment]
pushed https://github.com/dmick/teuthology/tree/wip-token with a suggested fix for these.  Comments welcome, raiding welcome

**Comment by dmick** [comment]
Filed #942 to extend/replace this

**Pull request closed without merging**